flink-streaming-connectors/flink-connector-hbase/src/main/java/org/apache/flink/streaming/connectors/hbase/HBaseMapper.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.connectors.hbase;
+
+import org.apache.flink.api.common.functions.Function;
+
+import java.io.Serializable;
+
+/**
+ * Maps a input value to a row in HBase table.
+ *
+ * @param <IN> input type
+ */
+public interface HBaseMapper<IN> extends Function, Serializable {
+
+	/**
+	 * Given an input value return the HBase row key. Row key cannot be null.
+	 *
+	 * @param value
+	 * @return row key
+	 */
+	byte[] rowKey(IN value);
```

Comment:
Hi @ramkrish86 , I was trying to and connect() method as you suggested and to explicitly test if table exists as following:
public void connect(String tableName) throws IOException {
		connection = ConnectionFactory.createConnection(hbConfig);
		TableName name = TableName.valueOf(tableName);
		try (Admin admin = connection.getAdmin()) {
			if (!admin.tableExists(name)) {
				throw new RuntimeException("Table " + tableName + " doesn't exist!");
			}
		}
		table = connection.getTable(name);
	}
But once the method admin.tableExists(name) is added, running my example will throw exception: org.apache.hadoop.hbase.regionserver.NoSuchColumnFamilyException: Column family table does not exist in region hbase:meta,,1.1588230740 in table 'hbase:meta'
I've never run into this exception in my project that uses hbase client. Do you know the reason for this exception?